### DISCUSSION TITLE
CBI Astute

### DISCUSSION THREAD

**Discussion by RubenvdM** [make-all/tuya-local#718] [General]
Anyone using the CBI Astute switches (Wall Plug, DB Board Switch)? I want to know if you get it better picked up as just a simple switch. I have edited the core.config_entries file and changed it to cbi_astute_outdoor_smartswitch which gives it a bit more functionality. 
However, I would love to have the timer enabled and also the initial state does not seem to work.

Any advice would greatly be appreciated.

Ruben

**Comment by make-all** [comment]
Don't edit the core.config_entries file to change a device type, that is a recipe for problems.
If a device config is not compatible with your device, you can't use it, file a new device request if your device does not match another config with all the functionality (but in the case of most smartswitches, there is probably a generic smartswitch_energy config that it matches).

**Reply by RubenvdM** [reply]
There is an existing device type for CBI Device type, however it is not picked up in discovery. I hear you on the core.config_entries file, and I agree. If I changed it, it works 85%, only the timer and initial state does not work. I am happy to log a new device req.
<img width="247" alt="image" src="https://github.com/make-all/tuya-local/assets/25762249/917ccfa3-ef1c-4cf2-9919-cba7d6f82cc8">

**Reply by make-all** [reply]
There are in fact two existing devices from this brand - an outdoor smartplug, and a circuit breaker energy monitor.  Other devices from the same brand are not necessarily compatible with these.

**Reply by RubenvdM** [reply]
yes and I have both, but they were not picked up in the discovery. Thats why I used the cbi_astute_outdoor_smartswitch.yaml file. I have logged a new device reg, but it is more likely and enhancement then.

**Reply by RubenvdM** [reply]
I just saw the update is there for the ones you said in the prev message. I deleted my old devices and re-added them. They are being recognized, however Initial state is now gone and timer is also not there.
<img width="247" alt="image" src="https://github.com/make-all/tuya-local/assets/25762249/f44dc46c-4237-4858-95a0-d07d97a6fec7">

in the Tuya app these are available for the device.
<img width="293" alt="image" src="https://github.com/make-all/tuya-local/assets/25762249/390f0997-20c8-46bc-a727-af9d329ecbdb">
